### PR TITLE
[CHORE] Bump localstack to latest

### PR DESCRIPTION
**Pull request by a6software**
We were using a very old version of Localstack, and a somewhat old version of the JS AWS-SDK. 

Amazon appears to have changed their implementation of the SQS protocol.

The only fix appears to be updating to the latest versions of localstack, JS AWS-SDK, and for test purposes the Java AWS-SDK. 

https://github.com/localstack/localstack/issues/8267#issuecomment-1806421471

and

https://mojdt.slack.com/archives/C69NWE339/p1699872586514569

--

hi @steverendell - requesting review as this contains a change to the Groovy AWS-SDK library usage. The behaviour is identical as best I am aware.

### CODE REVIEW

File: docker-compose-test.yml [hunk 1]
```diff
@@ -27,23 +27,22 @@ services:
...
-      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
+      - PERSISTENCE=${DATA_DIR- }
       - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
-      - DEFAULT_REGION=eu-west-2
-    volumes:
-      - './.localstack:/tmp/localstack'
+      - AWS_DEFAULT_REGION=eu-west-2
+      - LOCALSTACK_HOST=categorisation-localstack
+#    volumes:
+#      - './.localstack:/tmp/localstack'
```
Comment (a6software) [inline docker-compose-test.yml:45]: not sure if we need this? it kept a bunch of files for what seemed like debug purposes, but no issue with it being ephemeral as best i can tell

File: .circleci/config.yml [hunk 2]
```diff
@@ -79,11 +79,10 @@ jobs:
           POSTGRES_USER: form-builder
           POSTGRES_DB: form-builder
       - image: redis
-      - image: localstack/localstack:0.11.2
+      - image: localstack/localstack:latest
         environment:
           SERVICES: sqs
-          PORT_WEB_UI: 8088
```
Comment (a6software) [inline .circleci/config.yml:85]: removed in recent localstack

File: .circleci/config.yml [hunk 3]
```diff
@@ -79,11 +79,10 @@ jobs:
...
-          DATA_DIR: /tmp/localstack/data
+          PERSISTENCE: /tmp/localstack/data
```
Comment (a6software) [inline .circleci/config.yml:85]: renamed inline with updated image

File: .circleci/config.yml [hunk 4]
```diff
@@ -79,11 +79,10 @@ jobs:
           POSTGRES_USER: form-builder
           POSTGRES_DB: form-builder
       - image: redis
-      - image: localstack/localstack:0.11.2
+      - image: localstack/localstack:latest
```
Comment (a6software) [inline .circleci/config.yml:82]: using `latest` is the only way to get the fix. Need a follow on change to pin to a version once 3.0.0 drops.

File: docker-compose-test.yml [hunk 5]
```diff
@@ -27,23 +27,22 @@ services:
       - "6379:6379"
 
   categorisation-localstack:
-    image: localstack/localstack:0.11.2
+    image: localstack/localstack:latest
     ports:
-      - "4567-4584:4567-4584"
+      - "4566-4584:4566-4584"
```
Comment (steverendell) [inline docker-compose-test.yml:32]: You only need 4566:4566

**Review by steverendell — APPROVED** [review]
lgtm with minor comments (if it works then fine ! )

File: docker-compose-test.yml (hunk 5 above)
Comment (a6software) [reply docker-compose-test.yml]: trimmed, thanks

File: docker-compose-test.yml [hunk 6]
```diff
@@ -27,23 +27,22 @@ services:
...
-      - PORT_WEB_UI=${PORT_WEB_UI-8088}
-      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
+      - PERSISTENCE=${DATA_DIR- }
       - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
-      - DEFAULT_REGION=eu-west-2
+      - AWS_DEFAULT_REGION=eu-west-2
+      - LOCALSTACK_HOST=categorisation-localstack
     volumes:
-      - './.localstack:/tmp/localstack'
+      - "/var/run/docker.sock:/var/run/docker.sock"
```
[the remark this answers is not in the document]
Comment (a6software) [reply docker-compose-test.yml:47]: I've copied the lines now from the docs

kept the docker sock, added the localstack volume entry

https://docs.localstack.cloud/getting-started/installation/#starting-localstack-with-docker-compose

**Review by jonhindle — APPROVED** [review]
lgtm

**Pull request merged**